rosidl_generator_cpp/include/rosidl_generator_cpp/bounded_vector.hpp

```diff
@@ -0,0 +1,712 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSIDL_GENERATOR_CPP__BOUNDED_VECTOR_HPP_
+#define ROSIDL_GENERATOR_CPP__BOUNDED_VECTOR_HPP_
+
+#include <algorithm>
+#include <memory>
+#include <stdexcept>
+#include <utility>
+#include <vector>
+
+namespace rosidl_generator_cpp
+{
+
+/**
+ *  @brief  A container based on std::vector but with an upper bound.
+ *
+ *  @tparam  _Tp  Type of element.
+ *  @tparam  _UpperBound  The upper bound for the number of elements.
+ *  @tparam  _Alloc  Allocator type, defaults to allocator<_Tp>.
```

Comment:
This is a completely new class, regardless of whether or not you copied it from somewhere before modifying it. I think it would be better to use a consistent doc style, even in this case. What do others think?